### DISCUSSION TITLE
Using Devika in Local With Ollama

### DISCUSSION THREAD

**Discussion by gorbabor** [stitionai/devika#308] [Q&A]
I need this help on the configuration to do

**Comment by hqnicolas** [comment]
$ git clone https://github.com/hqnicolas/devika
$ cd devika
create config file
edit the docker compose to point to your ollama server url
$ sudo docker compose build
$ sudo docker compose up

**Reply by hqnicolas** [reply]
same as https://github.com/stitionai/devika/issues/257 #257 257

**Comment by ChanghongYangR** [comment]
![屏幕截图 2024-04-04 155152](https://github.com/stitionai/devika/assets/134147013/ca042940-5078-49f5-b1c6-e533edebc14e)
Any ideas why I couldn't get ollama work?

**Reply by hqnicolas** [reply]
$ git clone https://github.com/hqnicolas/devika
$ cd devika
create config file
edit the docker compose to ollama url
$ sudo docker compose build
$ sudo docker compose up

**Reply by gorbabor** [reply]
make it work, you'll need to launch Ollama as it's being described. I'll explain each step assuming you haven't installed Ollama yet. If that's the case:

Go to the Ollama website to download the version that suits you (I have Windows, so I downloaded the Windows version).
After downloading, install Ollama. By default, Ollama will start running on machine startup.
If that's not the case, you'll need to open a console (on Windows, launch the command prompt by typing "cmd" in the search bar) and enter the following command: "Ollama Serve." This will launch Ollama as a server, and it will provide its API through port 11434.

**Reply by cardchase** [reply]
you would need to open another terminal window. then 
cd devika/ui
or if you click the '+' and just
it should be comething like 
`~/devika$ `
then run
```
~/devika$ cd ui
~/devika/ui$ bun install 
```
then
`~/devika/ui$ bun run dev`
you should get

```
(base) user@user:~/devika/ui$ bun run dev
$ vite dev

  VITE v5.2.2  ready in 285 ms

  ➜  Local:   http://localhost:3000/
  ➜  Network: use --host to expose
  ➜  press h + enter to show help
```

a successful process would look like this:

![image](https://github.com/stitionai/devika/assets/33249394/853b3abd-2959-4bdd-b5a0-9457f870eab8)
I have run install and dev twice cause I had restarted the process but this is how you should be able to run.

It will be accessed via http://localhost:3000/ in your web browser

run these steps every time you have disconnected devika or started it

**Comment by gorbabor** [comment]
To make it work, you'll need to launch Ollama as it's being described. I'll explain each step assuming you haven't installed Ollama yet. If that's the case:

1. Go to the Ollama website to download the version that suits you (I have Windows, so I downloaded the Windows version).
2. After downloading, install Ollama. By default, Ollama will start running on machine startup.
3. If that's not the case, you'll need to open a console (on Windows, launch the command prompt by typing "cmd" in the search bar) and enter the following command: "Ollama Serve." This will launch Ollama as a server, and it will provide its API through port 11434.
4.

**Comment by phalexo** [comment]
Being able to access the API from ollama does NOT mean it will work. It will encounter all kinds of issues wth the local model itself, not producing the same output as GPT-4.

**Comment by trymeouteh** [comment]
Would like to see easy setup feature for Ollama

**Comment by bjornwed** [comment]
running dockerized version I had to set Ollama endpoint to "http://ollama-service:11434" as the container running ollama in devika-subnetwork will be accessible through the host "ollama-service".  After that all good, time to play with local llama3 :)

**Reply by rajeevchourasiya** [reply]
I need help in setup , can you please help me out.

**Reply by bjornwed** [reply]
> I need help in setup , can you please help me out. 

If you describe your issue and environment I can try. Although I'm not a devOp per say :)

**Reply by rajeevchourasiya** [reply]
Actually when i run devika, it is not browsing the web although i have entered my google API key, but not sure what to put in the endpoint.

**Reply by bjornwed** [reply]
The endpoint for Google should already be in the config.toml file, but if not it is: https://www.googleapis.com/customsearch/v1

So in Google you need to create a search engine, get the ID for the engine and create a key for your engine:
https://developers.google.com/custom-search/v1/introduction

To check everything's working correct on Google side, you can simply try the following in a browser:

`https://www.googleapis.com/customsearch/v1?key=[insert) your key]&cx=[insert your engine id]&q=lectures`

**Reply by rajeevchourasiya** [reply]
Thanks, but I want to use ollama. Although everything is set up perfectly, it just doesn't output anything, just stepwise plan at most sometimes, that's it.

**Reply by bjornwed** [reply]
You need to configure both, search engine for the agent to be able to do research for the project, and ollama or other LLM for agents to be able to perform tasks like generate code etc.
Also, depending on your hardware, response may be painfully slow if running everything local, so be patient.

**Reply by djpapzin** [reply]
Can you explain more what you mean

**Comment by albertotorresfoltyn** [comment]
still can;t run devika with local ollama

**Comment by djpapzin** [comment]
Have you tried OpenDevin?

**Reply by mchzimm** [reply]
i'd say that's out of scope here, this is about devika
